North-Seattle-College/ad440-winter2021-thursday-repo
# Adds React UI for creating a task

resolve #193 



This feature adds a form that allows the creation of a new task based on a given user. This added feature takes the form data except the User ID and creates corresponding JSON (using JS code) and passes this JSON as a body of the request to the API endpoint. The userId is implemented into the link - it is a dynamic of the API URL Endpoint of the API that is handled in CreateTask.js

## Testing

- Navigate to the form by pressing the Create Task button on the home page
- Fill out the form with the option of leaving the last two date fields null
- Submit the task and if the operation is successful you will see the notification that it was a success 

<img width="1265" alt="Screen Shot 2021-03-03 at 8 02 20 PM" src="https://user-images.githubusercontent.com/57120136/109910307-ab6fd300-7c5c-11eb-9f94-b70e46767c75.png">

- To change the userId simply go to the CreateTask.js page and swamp out the link with the chosen task link from your development.

<img width="652" alt="Screen Shot 2021-03-03 at 8 16 14 PM" src="https://user-images.githubusercontent.com/57120136/109911074-45844b00-7c5e-11eb-8838-ce4be9794731.png">

- The goal was to post to this API endpoint `https://nsc-fun-dev-usw2-thursday.azurewebsites.net/api/users/1/tasks` but I believe there is possibly a bug with this link as any number the user input is, the same two users show on the screen. Eventually this link should have create user input functionality  for the userId if this is further developed.

**Troubleshooting Tips**

There are 2 outputs from the console when creating a task, The first number is the userID that the task is going towards and the second field is the task in JSON  format. If these two dont show up it can be linked back to this functions and if a error shows up when creating the task there is a number that will indicate where the error should be.


Date | Activity | Time Spent
-- | -- | --
2/25/2021|Research form handling in React |3 hours
2/28/2021|Create Class version of the form  |2 hours
3/1/2021|Create a Functional version of the form | 2 hours 
3/2/2021|Implement userID function with POST | 3 hours 
3/3/2021|Fix bugs | 4 hours
